x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/types.ts

```diff
@@ -248,7 +267,7 @@ export interface ExportDocumentsProps {
   ids: string[];
   filename?: string;
   excludeExportDetails?: boolean;
-  signal: AbortSignal;
+  signal?: AbortSignal;
```

Comment:
Why is signal now optional? Does it need to be optional? This can lead to people not sending down the abort signal. If we can keep it required that would keep people more honest about making them have to send down the abort signal.

Reply:
That API call is used inside `getBatchItems()` (https://github.com/elastic/kibana/pull/100554/files/afd419abf31a9f7ea3a63dc8fb55e64aa208eeec#diff-6072a72c06d7a4afb6f77edafc82be6f1e0a427fd35909f06b978ab805011ea5) which is not a React component or hook. So passing down the abort signal wouldn't make sense because there is no lifecycle method to call `abort()`. 

There are five API calls in total without abortion logic, and I wouldn't say I like that either. I'll check if `getBatchItems()`  could be refactored into a hook so that we can add the abortion logic.

Reply:
Ok, thanks for the info, always curious for myself, you know. Usually good to know where we are starting to go "sideways" with regards to the code and why we are doing the things we are.

Reply:
Added it as a further improvement to the `RulesTables` refactoring ticket: https://github.com/elastic/kibana/issues/92160#issuecomment-856730693